kubectl-cloudflow/docker/client.go

```diff
@@ -147,53 +151,71 @@ func imageMatchesNameAndTag(image types.ImageSummary, imageNameAndTag string) bo
 		}
+	}
+}
 
-		// uncompressed data : []byte
-		uncompressed, err := ioutil.ReadAll(reader)
+// parseImageInspectData gets inspection data and the imageName and returns streamlet descriptors and other related info about the inspected image
+func parseImageInspectData(outputData InspectOutput, imageName string)(cloudflowapplication.CloudflowStreamletDescriptorsDigestPair, string, *PulledImage, error) {
+	label, found := outputData.Labels[streamletDescriptorsLabelName]
+	if found {
+		descriptors, err := getDescriptorsFromCompressedLabel(label)
 		if err != nil {
-			return cloudflowapplication.CloudflowStreamletDescriptorsDigestPair{}, "", nil, fmt.Errorf("Failed to read contents of the application descriptor, %s", err.Error())
+			return cloudflowapplication.CloudflowStreamletDescriptorsDigestPair{}, "", nil, fmt.Errorf("Failed to get streamlet descriptors from compressed Labelfor image: %s, %s", imageName, err.Error())
+		} else {
+			var descriptorsDigest cloudflowapplication.CloudflowStreamletDescriptorsDigestPair
+			descriptorsDigest.ImageDigest = outputData.Digest.String()
+			descriptorsDigest.StreamletDescriptors = descriptors.StreamletDescriptors
+			return descriptorsDigest, descriptors.APIVersion, &PulledImage{ImageName:imageName, Authenticated:true}, nil
 		}
+	} else {
+		return cloudflowapplication.CloudflowStreamletDescriptorsDigestPair{}, "", nil, fmt.Errorf("No label %s is found in local image: %s", streamletDescriptorsLabelName, imageName)
+	}
+}
 
-		// unmarshall to struct
-		var descriptors cloudflowapplication.Descriptors
-		json.Unmarshal([]byte(uncompressed), &descriptors)
-
-		descriptorsDigest.StreamletDescriptors = descriptors.StreamletDescriptors
-		reader.Close()
-		_, descriptorsDigest.ImageDigest = path.Split(image.RepoDigests[0])
-		return descriptorsDigest, descriptors.APIVersion, pulledImage, nil
+// getDescriptorsFromCompressedLabel gets the streamlet descriptors out of a compressed label
+func getDescriptorsFromCompressedLabel(label string)(*cloudflowapplication.Descriptors, error) {
+	compressed := base64.NewDecoder(base64.StdEncoding, bytes.NewReader([]byte(label)))
+	reader, err := zlib.NewReader(compressed)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to decompress the Cloudflow streamlet descriptors label, %s", err.Error())
+	}
+	uncompressed, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read the decompressed Cloudflow streamlet descriptors label %s", err.Error())
 	}
-	return cloudflowapplication.CloudflowStreamletDescriptorsDigestPair{}, "", nil, fmt.Errorf("Unable to inspect image '%s'. It could not be found locally", imageName) 
+	var descriptors cloudflowapplication.Descriptors
+	err = json.Unmarshal([]byte(uncompressed), &descriptors)
+
+	defer func() {
+		err := reader.Close()
+		if err != nil {
+			// most likely we dont want to fail here, skip the error
+			fmt.Println("Failed to close zlib reader")
```

Comment:
UX question: What is the user going to do with this message?

Reply:
It should be a debug message but we havent setup logging proerly for the cli. This should be improved when refactor the cli code base. I might remove it for now so it does not look weird.